cli/src/pcluster/config/cluster_config.py

```diff
@@ -1629,16 +1629,140 @@ def __init__(
+        if not self.__instance_types_with_instance_storage:
+            for instance_type in self.instance_types:
+                if self.instance_type_info_map[instance_type].instance_storage_supported():
+                    self.__instance_types_with_instance_storage.append(instance_type)
+        return self.__instance_types_with_instance_storage
+
+    @property
+    def instance_type_info_map(self) -> Dict[str, InstanceTypeInfo]:
+        """List of Instance Type information for each instance type in the compute resource.
+
+        :returns: [ "InstanceType1": {... InstanceTypeInfo ...}, "InstanceType1": {... InstanceTypeInfo ...} ]
+        """
+        if not self.__instance_type_info_map:
+            self.__instance_type_info_map = {
+                instance_type: self.fetch_instance_type_info(instance_type) for instance_type in self.instance_types
+            }
+        return self.__instance_type_info_map
+
+    @property
+    @abstractmethod
+    def disable_simultaneous_multithreading_manually(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def disable_simultaneous_multithreading_via_cpu_options(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def pass_cpu_options_in_launch_template(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def max_network_interface_count(self) -> int:
+        pass
+
+    @property
+    def is_ebs_optimized(self) -> bool:
+        if "unsupported" in [
+            self.instance_type_info_map[instance_type].ebs_optimized_support() for instance_type in self.instance_types
+        ]:
+            return False
+        return True
+
+    @property
+    @abstractmethod
+    def instance_types(self) -> List[str]:
+        pass
+
+
+class FlexibleInstanceType(Resource):
+    """Represent an instance type listed under InstanceTypes under ComputeResources."""
+
+    def __init__(self, instance_type: str, **kwargs):
+        super().__init__(**kwargs)
+        self.instance_type = Resource.init_param(instance_type)
+
+
+class SlurmFlexibleComputeResource(_BaseSlurmComputeResource):
+    """Represents a Slurm Compute Resource with Multiple Instance Types."""
+
+    def __init__(self, instance_type_list: List[FlexibleInstanceType], **kwargs):
+        super().__init__(**kwargs)
+        self.instance_type_list = Resource.init_param(instance_type_list)
+
+    @property
+    def instance_types(self) -> List[str]:
+        """Return list of instance type names in this compute resource."""
+        return [flexible_instance_type.instance_type for flexible_instance_type in self.instance_type_list]
+
+    @property
+    def disable_simultaneous_multithreading_manually(self) -> bool:
+        """Return true if simultaneous multithreading must be disabled with a cookbook script."""
+        return self.disable_simultaneous_multithreading
+
+    @property
+    def disable_simultaneous_multithreading_via_cpu_options(self) -> bool:
```

Comment:
this and following property have been removed as part of https://github.com/aws/aws-parallelcluster/pull/4260

Reply:
you can remove the `disable_simultaneous_multithreading_via_cpu_options` method